openstack/objectstorage/v1/objects/requests.go

```diff
@@ -611,8 +635,10 @@ func CreateTempURL(c *gophercloud.ServiceClient, containerName, objectName strin
 	}
 
 	secretKey := []byte(tempURLKey)
-	splitPath := strings.Split(url, opts.Split)
```

Comment:
incidental fix: `Split` splits `url` each time `opts.Split` (default: `/v1/`) is found. But below, we only joined the first two parts back into a full URL. Which means that a path like `https://swift/v1/AUTH_123/container/my/v1/object` became `https://swift/v1/AUTH_123/container/my` (see https://github.com/gophercloud/gophercloud/issues/2827)